### PR TITLE
Dashboard: Template Detail Redesign Updates 

### LINKED ISSUE #6596: Update dashboard template details view
Update dashboard template details view to use design system and match figma: https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=467%3A13745

### Acceptance Criteria
- No regressions in a11y usability of detail view 
- View should match updates in figma

### PR DESCRIPTION
**Pull request by BrittanyIRL**
## Context

Updates the template details view to new figma redesign specs: https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=467%3A13745 

## Summary

| Old      | New |
| ----------- | ----------- |
| Screen Shot 2021-03-03 at 10 08 06 AM | Screen Shot 2021-03-03 at 10 07 46 AM |
| Screen Shot 2021-03-03 at 10 07 06 AM | Screen Shot 2021-03-03 at 10 10 20 AM
 |

## Relevant Technical Choices

- Nothing weird here, mostly deleting dead code and simplifying a little bit. (only 9 / 33 files changed here have any actual updates, everything else is just pruning dead code)
- Deleted bookmark placeholder stuff, that's super dead and when we bring it back it'll be easier to just do it fresh.
- Made sure heading outline is accurate and added some aria labeling with template title to be more useful. 
Screen Shot 2021-03-03 at 9 51 54 AM

## To-do

- The dashboard grid PR I have up (#6588 ) has some consolidation of dashboard spacing values so when both of these are merged I can do a lot more clean up with static values here - like the padding and margin. 
- I'm not sure about the detail view spacing on smaller screens but that can be a follow up.  
- "Related Templates" isn't updated, it'll update with #6588 since it's using the shared `templateGrid`

## User-facing changes

- Design updates: Text elements, buttons, icons 

## Testing Instructions

- See that pagination still works as expected 
- See that 'use template' button still works as expected
- See that 'close' button still works as expected

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6596

### CODE REVIEW

File: assets/src/dashboard/app/views/templateDetails/components.js [hunk 1]
```diff
@@ -44,58 +45,7 @@ export const DetailContainer = styled.section.attrs({
 `;
 
 export const Column = styled.div`
-  ${({ theme }) => `
```
Comment (BrittanyIRL) [inline assets/src/dashboard/app/views/templateDetails/components.js]: To _try_ (and fail really) to keep the diff easier to read I only moved styled components out of this `templateDetails/components` when they were unnecessary or had a new base - like using design system stuff. Then I moved them more locally. This is a bit of a mixed bag and I'll clean it up better outside of these updates.

File: assets/src/dashboard/app/views/templateDetails/content/detailsGallery/index.js [hunk 2]
```diff
@@ -19,28 +19,53 @@
...
+  display: flex;
+  align-items: center;
+  ${({ rightPadding }) =>
+    rightPadding ? 'padding-right: 52px;' : 'padding-left: 52px;'}
+`;
+
+const TemplateTag = styled(Chip)`
+  margin-right: 12px;
+  > span {
+    color: ${({ theme }) => theme.colors.fg.primary} !important;
```
Comment (BrittanyIRL) [inline assets/src/dashboard/app/views/templateDetails/content/detailsGallery/index.js:60]: These are disabled chips but they aren't supposed to look like it so i'm overriding the span color

File: assets/src/dashboard/app/views/templateDetails/content/detailsGallery/index.js [hunk 3]
```diff
@@ -51,23 +76,38 @@ function DetailsGallery({
...
-      <PaginationButton
-        rotateRight
+      <Button
+        type={BUTTON_TYPES.TERTIARY}
+        size={BUTTON_SIZES.SMALL}
+        variant={BUTTON_VARIANTS.CIRCLE}
         aria-label={__('View previous template', 'web-stories')}
-        onClick={() => switchToTemplateByOffset(-1)}
+        onClick={({ currentTarget }) => {
+          currentTarget.blur();
```
Comment (BrittanyIRL) [inline assets/src/dashboard/app/views/templateDetails/content/detailsGallery/index.js:86]: This fixes an annoying issue where when you page left or right the button focus doesn't go away! Something to do with the buttons being memoized i think? If anyone has a better solution, I'm ready for it.

File: assets/src/dashboard/components/cardGallery/components.js [hunk 4]
```diff
@@ -56,34 +56,27 @@ export const ItemContainer = styled.div`
...
-    };
-    border-width: 2px; 
-
-    ${KEYBOARD_USER_SELECTOR} &:focus {
-      border-radius: 0;
-      border-color: ${rgba(theme.DEPRECATED_THEME.colors.bluePrimary, 0.85)};
-      outline: none;
-    }
+    border-radius: ${theme.borders.radius.small};
+    ${themeHelpers.focusableOutlineCSS(theme.colors.border.focus)};
```
Comment (embarks) [inline assets/src/dashboard/components/cardGallery/components.js:79]: nice, so useful

File: assets/src/dashboard/app/views/templateDetails/header/index.js [hunk 5]
```diff
@@ -18,33 +18,78 @@
...
+          as="a"
+          aria-label={sprintf(
+            /* translators: %s: page title of link */
+            __('Go to %s', 'web-stories'),
+            ROUTE_TITLES[APP_ROUTES.TEMPLATES_GALLERY]
+          )}
+          href={parentRoute()}
+        >
+          <Icons.Cross />
+        </Button>
```
Comment (samwhale) [inline assets/src/dashboard/app/views/templateDetails/header/index.js]: I think this needs to be a little bigger. It's hard to tell from these screenshots but I think the icon itself should be close to 32px, not the bounding button

The button looks to be around 56px: 



|Current|Designs|
|--|--|
|image|image|

File: assets/src/dashboard/app/views/templateDetails/content/detailsGallery/index.js [hunk 6]
```diff
@@ -95,46 +135,47 @@ function DetailsGallery({
...
+
+          <MetadataContainer>
+            {template.tags.map((tag) => (
+              <TemplateTag key={tag} disabled>
+                {tag}
+              </TemplateTag>
+            ))}
+          </MetadataContainer>
+          <MetadataContainer>
+            <ColorList colors={template.colors} size={30} />
```
Comment (samwhale) [inline assets/src/dashboard/app/views/templateDetails/content/detailsGallery/index.js]: I think the colors need to be 32 px 
image

File: assets/src/dashboard/components/cardGallery/components.js [hunk 7]
```diff
@@ -97,7 +90,9 @@ export const MiniCard = styled.div(
     position: relative;
     width: ${width}px;
     cursor: pointer;
-    border: ${theme.DEPRECATED_THEME.borders.gray75};
+    border: 1px solid ${theme.colors.border.defaultNormal};
+    border-radius: ${theme.borders.radius.small};
+
```
Comment (samwhale) [inline assets/src/dashboard/components/cardGallery/components.js:95]: Are the mini cards supposed to be updated here too? It looks like they need to be a little bigger

|Current|Designs|
|--|--|
|image|image|

File: assets/src/dashboard/app/views/templateDetails/content/relatedGrid/index.js [hunk 8]
```diff
@@ -38,7 +39,10 @@ function RelatedGrid({ pageSize, relatedTemplates, templateActions }) {
 
   return (
     <RowContainer>
-      <SubHeading>{__('Related Templates', 'web-stories')}</SubHeading>
+      <Headline size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL} as="h3">
+        {__('Related Templates', 'web-stories')}
+      </Headline>
+
```
Comment (samwhale) [inline assets/src/dashboard/app/views/templateDetails/content/relatedGrid/index.js:45]: Looks like this needs a little more space

|Current|Designs|
|--|--|
|image|image|

File: assets/src/dashboard/app/views/templateDetails/content/index.js [hunk 9]
```diff
@@ -65,20 +61,18 @@ function Content({
 
   return (
     <Layout.Scrollable>
-      <DetailViewContentGutter>
-        <DetailsGallery
-          activeTemplateIndex={activeTemplateIndex}
```
Comment (samwhale) [inline assets/src/dashboard/app/views/templateDetails/content/index.js:70]: nice catch

File: assets/src/dashboard/components/cardGallery/components.js [hunk 10]
```diff
@@ -111,7 +106,8 @@ export const ActiveCard = styled.div(
     width: ${width}px;
     overflow: hidden;
     height: ${containerHeight}px;
-    border: ${theme.DEPRECATED_THEME.borders.gray75};
+    border: 1px solid ${theme.colors.border.defaultNormal};
+    border-radius: ${theme.borders.radius.small};
```
Comment (samwhale) [inline assets/src/dashboard/components/cardGallery/components.js:110]: Looks like the active card should be bigger?

|Current|Designs|
|--|--|
|image|image|

File: assets/src/dashboard/components/index.js [hunk 11]
```diff
@@ -25,12 +24,7 @@ export {
   default as CardGridItem,
 } from './cardGridItem';
 export { default as ColorList } from './colorList';
-export {
-  DetailViewContentGutter,
-  StandardViewContentGutter,
-} from './contentGutter';
-export { DetailViewNavBar } from './detailViewNavBar';
-export { default as Dropdown } from './dropdown';
```
Comment (samwhale) [inline assets/src/dashboard/components/index.js:33]: niceeee

**Review by samwhale — COMMENTED** [review]
Looks super good so far. Left you a few questions, love the removal of all that dead code!

Other questions:
1. Should this 'bookmark' button/thing be added here or is it a different ticket? Feels like new functionality so it should probably be a different ticket.
image
2. Should the `Label`s here be tooltips, filled in, or nothing?
image

File: assets/src/dashboard/app/views/templateDetails/header/index.js (hunk 5 above)
Comment (BrittanyIRL) [reply assets/src/dashboard/app/views/templateDetails/header/index.js]: I swapped in the large cross instead, i think this fixes it.

**Review by samwhale — COMMENTED** [review]
Oh an the designs have a purple focus border, do we need to update the mini cards?

|Current|Designs|
|--|--|
|image|image|

File: assets/src/dashboard/app/views/templateDetails/content/detailsGallery/index.js (hunk 6 above)
Comment (BrittanyIRL) [reply assets/src/dashboard/app/views/templateDetails/content/detailsGallery/index.js]: thanks! good catch.

File: assets/src/dashboard/components/cardGallery/components.js (hunk 7 above)
Comment (BrittanyIRL) [reply assets/src/dashboard/components/cardGallery/components.js:95]: this is a tricky one. The templates and stories have to be kept at the right ratio of a story (this util: `getPagePreviewHeights  ... Math.round((width / PAGE_RATIO) * 100) / 100;` ) so this gallery updates responsively according to how much room there is. 

I would rather handle any necessary updates to this as its own thing? 😬 


https://user-images.githubusercontent.com/10720454/110037252-60f16380-7cfb-11eb-951f-b7aa3e9d19b8.mp4

File: assets/src/dashboard/app/views/templateDetails/content/relatedGrid/index.js (hunk 8 above)
Comment (BrittanyIRL) [reply assets/src/dashboard/app/views/templateDetails/content/relatedGrid/index.js:45]: yup 🤦  updated the element to an h3 and it lost its style inheritance.
Screen Shot 2021-03-04 at 3 09 28 PM

File: assets/src/dashboard/components/cardGallery/components.js (hunk 10 above)
Comment (BrittanyIRL) [reply assets/src/dashboard/components/cardGallery/components.js:110]: samsies as the above with grid resizing. I think this should be separate and follow up.

**Comment by BrittanyIRL** [comment]
> Oh an the designs have a purple focus border, do we need to update the mini cards?

Think this is just a neglected update. Focus styles should match what we're doing elsewhere with the blue.

**Comment by BrittanyIRL** [comment]
Thank you for being so attentive and inquisitive, @samwhale !

Just pushed up changes for everything _except_ grid aspect ratio, which we should do separate 😅 . I will make a ticket this week. 

> * Should this 'bookmark' button/thing be added here or is it a different ticket? Feels like new functionality so it should probably be a different ticket.

Since bookmark isn't a thing we can do in the dashboard (yet) I decide to just take the axe to the placeholder since there was so much rewrite happening. 

> * Should the `Label`s here be tooltips, filled in, or nothing?

I didn't know what should go on them so I decided to skip them until we can get design input. There's no regression since this isn't something that currently exists and everything has proper tagging for SR.

**Review by embarks — APPROVED** [review]

**Pull request merged**